lib/THNN/generic/SpatialAdaptiveAveragePooling.c

```diff
@@ -0,0 +1,249 @@
+#ifndef TH_GENERIC_FILE
+#define TH_GENERIC_FILE "generic/SpatialAdaptiveAveragePooling.c"
+#else
+
+static void THNN_(SpatialAdaptiveAveragePooling_updateOutput_frame)(
+          real *input_p,
+          real *output_p,
+          long nslices,
+          long iwidth,
+          long iheight,
+          long owidth,
+          long oheight,
+          long stridew,
+          long strideh,
+          long strided)
+{
+  long k;
+#pragma omp parallel for private(k)
+  for (k = 0; k < nslices; k++)
+  {
+    /* loop over output */
+    long i, j;
+    for(i = 0; i < oheight; i++)
+    {
+      int y_start = (int)floor((float)i / oheight * iheight);
+      int y_end   = (int)ceil((float)(i + 1) / oheight * iheight);
+      int kH = y_end-y_start;
```

Comment:
Is it better to use
int y_start = (int)floor(float(i * iheight) / oheight);
int y_end   = (int)ceil((float)((i + 1) * iheight) / oheight);
or 
int y_start = i * iheight / oheight;
int y_end = i * iheight / oheight + (i * iheight % oheight)>0?1:0;

The first one is faster and more accurate than the original implementation (but still involves float number calculation). The second is uglier but accurate.